tests/end2end/cypress/integration/legend-ghaction.js

```diff
@@ -34,13 +34,20 @@ describe('Legend tests', function () {
                 const responseBodyAsBase64 = arrayBufferToBase64(interception.response.body)
 
                 let expected_path = 'images/treeview/' + check +'.png'
-                if (Cypress.env('QGIS_SERVER_INT') <= 31600 && check == 'layer_legend_categorized'){
-                    expected_path = 'images/treeview/' + check +'_316.png'
-                }
-
-                cy.fixture(expected_path).then((image) => {
-                    expect(image, 'expect legend to be displayed').to.equal(responseBodyAsBase64)
-                })
+                serverMetadata().then(metadataResponse => {
```

Comment:
test the `check` before calling serverMetadata to prevent usesless call ? 
``` 
if (check == 'layer_legend_categorized') {
  serverMetadata().then(metadataResponse => {
        if (metadataResponse.body.qgis_server_info.metadata.version_int <= 31600 ) {
            expected_path = 'images/treeview/' + check +'_316.png';
        }
    });
 }
 ```
and maybe chaining ` then` will work  ?
```` 
 serverMetadata().then(metadataResponse => {
        if (metadataResponse.body.qgis_server_info.metadata.version_int <= 31600 ) {
            expected_path = 'images/treeview/' + check +'_316.png';
        }
    }).then(metadataResponse => {
         if (metadataResponse.body.qgis_server_info.metadata.version_int <= 32800) {
             // With QGIS 3.28 : https://github.com/qgis/QGIS/pull/50256
             cy.fixture(expected_path).then((image) => {
                  expect(image, 'expect legend to be displayed').to.equal(responseBodyAsBase64)
        })
 };

Reply:
Yes, i can have a look.

The best would be to have a function sync like :  

```javascript
it('should print title labels (PNG)', function () {
    let qgisVersion = serverMetadata();
    // blabla
    let diff = 0;
    if qgisVersion < 32800:
        diff = 230;
    /foo
```

Reply:
Fixed indeed, it's better, thanks